### PR TITLE
[THORN-2496] modernize the Thorntail generator

### PR DESCRIPTION
**Pull request by Ladicek**
This commit modernizes the Thorntail generator all around the place.
Specifically:

- The application itself is no longer built on an outdated version
  of WildFly Swarm, but on latest Thorntail.
- It is possible to specify the version string of the generated
  application, as well as an explicit package name instead of
  the combination of `groupId` and `artifactId`. None of that
  is mandatory; if missing, the original behavior is retained.
- It is also possible to specify that the generated archive should
  not contain a top-level directory, but directly the generated
  application. This is off by default to preserve original behavior.
  It is added for the purpose of Halkyon, which expects this flat
  (non-nested) style of archives.
- Few other tiny improvements and cleanups.

### CODE REVIEW

**Comment by Ladicek** [comment]
I wonder where the generator is hosted. @gastaldi I think you've been running it somewhere?

**Comment by gastaldi** [comment]
Hm, I have no idea either. I think @bobmcwhirter hosts it in a dedicated server somewhere?

File: src/test/java/org/swarm/generator/rest/EndpointFilePathGeneratorWithExplicitPackageTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package org.swarm.generator.rest;
```
Comment (gastaldi) [inline src/test/java/org/swarm/generator/rest/EndpointFilePathGeneratorWithExplicitPackageTest.java]: Maybe move to the `io.thorntail.generator` package?

**Review by gastaldi — APPROVED** [review]
LGTM. I suggested moving `org.swarm.generator` to `io.thorntail.generator` but that's not a big deal :)

File: src/test/java/org/swarm/generator/rest/EndpointFilePathGeneratorWithExplicitPackageTest.java (hunk 1 above)
Comment (Ladicek) [reply src/test/java/org/swarm/generator/rest/EndpointFilePathGeneratorWithExplicitPackageTest.java]: Good point, I shall do that :-)

**Comment by Ladicek** [comment]
Ah, OK, thanks! I'll ask.

**Pull request merged**